shared-module/audiofilters/Filter.c

```diff
@@ -240,8 +240,21 @@ audioio_get_buffer_result_t audiofilters_filter_get_buffer(audiofilters_filter_o
             }
         }
 
-        // If we have a sample, filter it
-        if (self->sample != NULL) {
+        if (self->sample == NULL) {
+            if (self->samples_signed) {
+                memset(word_buffer, 0, length * (self->bits_per_sample / 8));
+            } else {
+                // For unsigned samples set to the middle which is "quiet"
+                if (MP_LIKELY(self->bits_per_sample == 16)) {
+                    memset(word_buffer, 32768, length * (self->bits_per_sample / 8));
```

Comment:
I don't know that this is right. Isn't memset always working in byte units, in which case this is likely to fill with all zeros (the lower 8 bits of 32768)?

Reply:
Good find. Similar code is being used with `audiodelays.Echo` where I pulled this from. We'll need to look into a better solution.

Reply:
Just implemented the fix using a while loop. It won't be as performant as `memset`, but I couldn't find any alternative. Here's the script I am using to diagnose the output. https://gist.github.com/relic-se/1bf0f95aaf444e4d5a0fd47228c3bb31

I updated `audiodelays.Echo` as well. In order to test that, you can swap out the module in the prior gist and set `mix=0.0`. The resulting output is the same.